cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -219,6 +297,14 @@ public boolean containsKey(String key) {
     return Collections.unmodifiableList(new ArrayList<>(details.keySet()));
```

Comment:
because details is unmodifiable, do you need to copy?

Reply:
also what is getValues() needed for? If it is not needed, let's remove it.

Reply:
Fixed the defensive copying.
getValues() is useful in testing. It has not been in used in non test code yet but it is convenient method to have.